src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx

```diff
@@ -39,36 +32,42 @@ const defaultClickedState = (): ClickedNodeState => {
     return null;
 };
 
+async function updateGraphAndSetState(
+    lensName: any,
+    state: any,
+    setState: any
+) {
+    if (lensName) {
+        await updateGraph(lensName, state as GraphState, setState); // state is safe cast, check that lens name is not null
+
+        console.log(
+            "setState",
+            await updateGraph(lensName, state as GraphState, setState)
+        );
+    }
+}
+
 const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
     const fgRef: any = useRef(); // fix graph to canvas
     const [state, setState] = useState(defaultGraphDisplayState(lensName));
-
-    async function updateGraphAndSetState() {
-        if (lensName) {
-            await updateGraph(lensName, state as GraphState, setState); // state is safe cast, check that lens name is not null
-        }
-    }
+    const [clickedNode, setClickedNode] = useState(defaultClickedState());
+    const [highlightNodes, setHighlightNodes] = useState(new Set());
+    const [highlightLinks, setHighlightLinks] = useState(new Set());
```

Comment:
Do we have types for a "Node" or a "Link"? If so: 

`useState(new Set<Node>());` 

and such

Reply:
I'll resolve this in a separate PR - we'll have to rework a couple of the types, making an issue, thanks for bringing this to my attention!